### ISSUE TITLE
Texcat accuracy settling on 0.5 while loss goes to 0 for binary classification

### ISSUE THREAD

**Issue by tsoernes** [explosion/spaCy#4720]
I've mimicked the example in the docs, and for a binary classification problem used two output neurons with `exclusive_classes=True` and labels as such:
```
{
        'cats': {
            'POS': False,
            'NEG': True
        }
    }
```
.. and architecture either `bow` or `ensemble`. I observe a strange issue:
After training 80k examples for an epoch, the loss, as reported by Spacy (from `nlp.update`), is `0.0000`, while the accuracy, as reported by scikit-learn's `accuracy_score` tends toward `0.5000` exactly and the  `log_loss` just keeps increasing.
```
from sklearn.metrics import accuracy_score, log_loss

docs = nlp.pipe(valid_texts, batch_size=256)
pred_probs = []
for doc in docs:
    pred_probs.append(list(doc.cats.values()))
y_pred_probs = np.array(pred_probs)

avg_log_loss = log_loss(y_true, y_pred_probs)
y_pred = y_pred_probs > 0.5
acc = accuracy_score(y_true, y_pred)
```
I've made sure the the order of Spacy's internal labels (as given by `doc.cats.keys()`) is the same as  the one-hot encoded numpy array `y_true`, and that `y_pred` and `y_true` are the same shape, that is, `(n_examples, 2)`. Evaluations are done using `with nlp.use_params(optimizer.averages)`. 

These results seems highly suspect to me and I'm not sure where I've stepped wrong. I'm using the `en_core_web_lg` model; disabling any pipes besides textcat, and training from scratch (i.e. `nlp.begin_training()`) without word vectors.

The label distribution is *exactly* 50-50 for both the training and the test set. 

Thanks for any advice, 

## Your Environment
* **spaCy version:** 2.2.3
* **Platform:** Linux-4.4.0-1098-aws-x86_64-with-debian-stretch-sid
* **Python version:** 3.7.4

**Comment by svlandeg** [comment]
Hi @tsoernes, this type of question is probably more suited for Stackoverflow as it's not immediately clear whether there's an issue in the spaCy code base. There always could be, ofcourse. Other possibilities are that there's a problem with the way you're interpreting/evaluating the performance, or there could be an issue in your data / project setup.

Some ideas to try out, from the top of my head:
- Inspect the training loss curve. If this looks like a genuine loss curve that gradually decreases over epochs, the spaCy training part seems to be going fine.
- Run the performance evaluation (with scikit-learn) after each epoch on the **training** data. The training performance (obviously) should go up. You can also output the performance on the test data, ofcourse, which should also go up.
- If either of them don't go up, inspect the actual predictions. Are they by any chance all POS or all NEG ?
- Reimplement the calculation of accuracy in your own code to debug what is going on

If the problem persists and you do think it is caused by the spaCy algorithms, feel free to create a new issue. When doing so, it would be helpful to create a minimal working code snippet + example data so that we can try and replicate, and debug, the problem.
